test/swar/BasicOperations.cpp

```diff
@@ -323,3 +337,76 @@ constexpr auto aBooleansWithTrue = booleans(SWAR<4, u32>{0x1});
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0111 (7) + 0x0001 (1) is 0x1000 (-8)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'8000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'F000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0101 (5) + 0x0101 (5) is 0x1010 (-x)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'5000), SWAR<4, u32>(0x0000'5000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'A000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0111 (7) + 0x0111 (7) is 0x1110 (0x1110->0x1101->0x0010) (e unsigned, 2 signed)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'7000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'e000).value() == fplusone.result.value());
+    }
+    {
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'a000), SWAR<4, u32>(0x0000'a000));
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+    }
+}
+
+TEST_CASE(
+    "booleanSWARasMask",
+    "[swar]"
+) {
+    // BooleanSWAR as a mask: 
+    auto bswar =BooleanSWAR<4, u32>(0x0808'0000);
+    auto mask = S4_32(0x0F0F'0000);
+    CHECK(bswar.asMask().value() == mask.value());
+}
+
+static_assert( S4_32(0x1111'1111).value() == fullAddition(S4_32(0x0111'1101), S4_32(0x1000'0010)).result.value());
+static_assert( S4_32(0x0000'0000).value() == fullAddition(S4_32(0x0111'1101), S4_32(0x1000'0010)).carry.value());
+static_assert( S4_32(0x0000'0000).value() == fullAddition(S4_32(0x0111'1101), S4_32(0x1000'0010)).overflow.value());
+
+static_assert( S4_16(0x0000).value() == saturatingUnsignedArithmetic(S4_16(0x0000), S4_16(0x0000)).value());
+
+static_assert( S4_16(0x0200).value() == saturatingUnsignedArithmetic(S4_16(0x0100), S4_16(0x0100)).value());
+static_assert( S4_16(0x0400).value() == saturatingUnsignedArithmetic(S4_16(0x0300), S4_16(0x0100)).value());
+static_assert( S4_16(0x0A00).value() == saturatingUnsignedArithmetic(S4_16(0x0300), S4_16(0x0700)).value());
+static_assert( S4_16(0x0F00).value() == saturatingUnsignedArithmetic(S4_16(0x0800), S4_16(0x0700)).value());
+static_assert( S4_16(0x0F00).value() == saturatingUnsignedArithmetic(S4_16(0x0800), S4_16(0x0800)).value());
```

Comment:
Is the reader supposed to infer the meaning of these tests by analysis of the constants?

Reply:
I learned it from watching you.

Reply:
1. Do as I say, not as I do ;-)
2. Seriously, I got legitimate flak for the anti-practice of plopping cryptic constants.  We've reached a level of quality in the actual mechanisms that I believe demands more of the rest
3. This is an example of "it costs the author little but may help the user/reader a lot"
4. I believe that we can devise a better way to express these magical constants, for example, defining named bitfields that match the lane size.
5. Perhaps paying attention to the constants we may discover some useful pattern.